extensions/BugModal/lib/WebService.pm

```diff
@@ -87,19 +87,24 @@ sub products {
     return { products => _name($user->get_enterable_products) };
 }
 
-sub components {
+sub product_info {
     my ( $self, $params ) = @_;
     if ( !ref $params->{product_name} ) {
         untaint( $params->{product_name} );
     }
     else {
-        ThrowCodeError( 'params_required',
-            { function => 'BugModal.components', params => ['product'] } );
+        ThrowCodeError( 'params_required', { function => 'BugModal.components', params => ['product'] } );
     }
-    my $product = Bugzilla::Product->check({ name => $params->{product_name}, cache => 1 });
-    $product = Bugzilla->user->can_enter_product($product, 1);
-    my @components = map { { name => $_->name, description => Bugzilla::Component->check({ product => $product, name => $_->name })->description} } @{ $product->components };
-    return { components => \@components }
+    my $product = Bugzilla::Product->check( { name => $params->{product_name}, cache => 1 } );
+    $product = Bugzilla->user->can_enter_product( $product, 1 );
+    my @components = map {
+        {
+            name        => $_->name,
+            description => $_->description,
+        }
+    } @{ $product->components };
+    my @versions = map { { name => $_->name } } grep( $_->is_active, @{ $product->versions } );
```

Comment:
so both grep and map have two forms: `grep code, list` and `grep { code } list`
It's better to just always use the block style.
so make this:
```perl
my @versions = map { { name => $_->name } } grep { $_->is_active } @{ $product->versions };